ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java

```diff
@@ -265,6 +265,65 @@ public PackageCompilation getCompilation(Collection<Snippet> newSnippets) throws
         return compilation;
     }
 
+    @Override
+    public PackageCompilation getCompilation(Snippet newSnippet) throws InvokerException {
+        if (!this.initialized.get()) {
+            throw new IllegalStateException("Invoker execution not initialized.");
+        }
+
+        newImports.clear();
+
+        // TODO: (#28036) Fix the closure bug.
+
+        variableDeclarations = new HashMap<>();
+        moduleDeclarations = new HashMap<>();
+        executableSnippets = new ArrayList<>();
+        variableNames = new ArrayList<>();
+
+        if (newSnippet instanceof ImportDeclarationSnippet) {
+            processImport((ImportDeclarationSnippet) newSnippet);
+
+        } else if (newSnippet instanceof VariableDeclarationSnippet) {
+            VariableDeclarationSnippet varDclnSnippet = (VariableDeclarationSnippet) newSnippet;
+            variableNames.addAll(varDclnSnippet.names());
+            variableDeclarations.put(varDclnSnippet, varDclnSnippet.names());
+
+        } else if (newSnippet instanceof ModuleMemberDeclarationSnippet) {
+            ModuleMemberDeclarationSnippet moduleDclnSnippet = (ModuleMemberDeclarationSnippet) newSnippet;
+            Identifier moduleDeclarationName = moduleDclnSnippet.name();
+            moduleDeclarations.put(moduleDeclarationName, moduleDclnSnippet);
+            Set<Identifier> usedPrefixes = newSnippet.usedImports().stream()
+                    .map(Identifier::new).collect(Collectors.toSet());
+            newImports.put(moduleDeclarationName, usedPrefixes);
+
```

Comment:
Here, as well

Reply:
Removed in ea054445b72a422651c331b20cc9c2bb6056a8f8